client/src/legacy/LeftAndMain.js

```diff
@@ -1508,7 +1508,13 @@ $.entwine('ss', function($) {
   });
 });
 
+// TODO: move this to an external util
+function decodeEntities(encodedString) {
+    var textArea = document.createElement('textarea');
+    textArea.innerHTML = encodedString;
+    return textArea.value;
+}
+
 var statusMessage = function(text, type) {
-  text = jQuery('<div/>').text(text).html(); // Escape HTML entities in text
-  jQuery.noticeAdd({text: text, type: type, stayTime: 5000, inEffect: {left: '0', opacity: 'show'}});
+  jQuery.noticeAdd({text: decodeEntities(text), type: type, stayTime: 5000, inEffect: {left: '0', opacity: 'show'}});
```

Comment:
On line 191 we have a message that is being passed into a `decodeURIComponent` function - it seems likely there's a risk of double-decoding there, if that's even a thing. Should we remove the call to `decodeURIComponent` on that line?
```js
statusMessage(decodeURIComponent(msg), msgType);
```

Reply:
The same thing is happening in https://github.com/silverstripe/silverstripe-admin/blob/4a97f0bd0f5d297684b81311353ba33573ac360c/client/src/legacy/LeftAndMain.BatchActions.js#L303

Reply:
as far as i'm aware, there is only a risk of double encoding, not double decoding but i may be wrong here
at any rate, "it seems to work" on my end with my changes. i know that's not a very satisfactory answer :-)

Reply:
I guess the thing I'm worried about (and it is _an extreme_ edge case so maybe we can ignore it? I'd want feedback from others in @silverstripe/core-team though) is if someone intentionally wants the encoded form.

e.g. a page called "`Are &quot; and `&#39;` the same thing?"

Reply:
i'm not sure either, i just did what needed in order to fix all my use cases :)

Reply:
I've just tried creating a page called `Are &quot; and ' and &#39; the same thing?` both with this (and the framework) PR, and without them.
In both cases, the toast displays like this:
![Screenshot from 2024-01-19 12-08-34](https://github.com/silverstripe/silverstripe-admin/assets/36352093/18b955a7-119c-4552-a7b2-e3411e2df2aa)

So this PR is still not handling some characters (`&` specifically in this case) correctly.
Can you please look into that?